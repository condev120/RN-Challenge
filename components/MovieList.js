import { FlatList, RefreshControl, StyleSheet } from "react-native";
import { MovieCard } from "./MovieCard";

export const MovieList = ({movies, full, loading, favoriteMovies, onRefresh, onFavoriteRefresh}) => (
  <FlatList
    testID="MovieList"
    refreshControl={<RefreshControl refreshing={loading} onRefresh={onRefresh} />}
    numColumns={full ? 1 : 2}
    data={movies}
    keyExtractor={(item) => item?.id}
    renderItem={({item}) => <MovieCard full={full} favoriteMovies={favoriteMovies} onRefresh={onRefresh} onFavoriteRefresh={onFavoriteRefresh} {...item}/>}
    contentContainerStyle={styles.contentContainer}
    columnWrapperStyle={!full && styles.contentContainerFull}
  />
);
const styles = StyleSheet.create({
  contentContainer: {
    padding: 20,
  },
  contentContainerFull: {
    flex: 1,
    justifyContent: 'space-between',
  },
})