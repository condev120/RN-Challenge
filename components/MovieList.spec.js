import React from 'react';
import { View } from 'react-native';
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import { render, waitFor } from '@testing-library/react-native';
import { mockMovies } from '../services/mock.service';
import { MovieList } from './MovieList';

let rendered;
jest.mock('react-native-ratings', () => ({
  AirbnbRating: jest.fn()
}))
jest.mock('react-native-vector-icons/Ionicons', () => jest.fn());
jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);
const movies = mockMovies.results;
describe('MovieList - Coverage', () => {
  const mockProps = {
    movies,
    full: true
  };
  beforeEach(async () => {
    await waitFor(async () => {
      rendered = render(<MovieList {...mockProps}/>);
    });
  });

  it('Renders correctly', async () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('Render movie list correctly', () => {
    expect(rendered.getByTestId('MovieList')).toBeDefined();
    expect(rendered.getByTestId('MovieList').props.data).toHaveLength(4);
  });
});
