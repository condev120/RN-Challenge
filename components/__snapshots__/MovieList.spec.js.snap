// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList - Coverage Renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "padding": 20,
    }
  }
  data={
    [
      {
        "adult": false,
        "backdrop_path": "/gMJngTNfaqCSCqGD4y8lVMZXKDn.jpg",
        "genre_ids": [
          "Adventure",
          "Animation",
        ],
        "id": 640146,
        "original_language": "en",
        "original_title": "Ant-Man and the Wasp: Quantumania",
        "overview": "Super-Hero partners Scott Lang and Hope van Dyne, along with with Hope's parents Janet van Dyne and Hank Pym, and Scott's daughter Cassie Lang, find themselves exploring the Quantum Realm, interacting with strange new creatures and embarking on an adventure that will push them beyond the limits of what they thought possible.",
        "popularity": 8567.865,
        "poster_path": "/ngl2FKBlU4fhbdsrtdom9LVLBXw.jpg",
        "release_date": "2023-02-15",
        "title": "Ant-Man and the Wasp: Quantumania",
        "video": false,
        "vote_average": 6.5,
        "vote_count": 1886,
      },
      {
        "adult": false,
        "backdrop_path": "/iJQIbOPm81fPEGKt5BPuZmfnA54.jpg",
        "genre_ids": [
          "Adventure",
          "Animation",
        ],
        "id": 502356,
        "original_language": "en",
        "original_title": "The Super Mario Bros. Movie",
        "overview": "While working underground to fix a water main, Brooklyn plumbers—and brothers—Mario and Luigi are transported down a mysterious pipe and wander into a magical new world. But when the brothers are separated, Mario embarks on an epic quest to find Luigi.",
        "popularity": 6572.614,
        "poster_path": "/qNBAXBIQlnOThrVvA6mA2B5ggV6.jpg",
        "release_date": "2023-04-05",
        "title": "The Super Mario Bros. Movie",
        "video": false,
        "vote_average": 7.5,
        "vote_count": 1456,
      },
      {
        "adult": false,
        "backdrop_path": "/nDxJJyA5giRhXx96q1sWbOUjMBI.jpg",
        "genre_ids": [
          "Adventure",
          "Horror",
        ],
        "id": 594767,
        "original_language": "en",
        "original_title": "Shazam! Fury of the Gods",
        "overview": "Billy Batson and his foster siblings, who transform into superheroes by saying "Shazam!", are forced to get back into action and fight the Daughters of Atlas, who they must stop from using a weapon that could destroy the world.",
        "popularity": 4274.232,
        "poster_path": "/2VK4d3mqqTc7LVZLnLPeRiPaJ71.jpg",
        "release_date": "2023-03-15",
        "title": "Shazam! Fury of the Gods",
        "video": false,
        "vote_average": 6.9,
        "vote_count": 1231,
      },
      {
        "adult": false,
        "backdrop_path": "/ovM06PdF3M8wvKb06i4sjW3xoww.jpg",
        "genre_ids": [
          "Adventure",
          "Horror",
        ],
        "id": 76600,
        "original_language": "en",
        "original_title": "Avatar: The Way of Water",
        "overview": "Set more than a decade after the events of the first film, learn the story of the Sully family (Jake, Neytiri, and their kids), the trouble that follows them, the lengths they go to keep each other safe, the battles they fight to stay alive, and the tragedies they endure.",
        "popularity": 3365.913,
        "poster_path": "/t6HIqrRAclMCA60NsSmeqe9RmNV.jpg",
        "release_date": "2022-12-14",
        "title": "Avatar: The Way of Water",
        "video": false,
        "vote_average": 7.7,
        "vote_count": 7535,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={<RefreshControlMock />}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  testID="MovieList"
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          [
            {
              "justifyContent": "flex-start",
              "marginBottom": 16,
            },
            {
              "flexDirection": "row",
            },
          ]
        }
        testID="MovieCard"
      >
        <View
          style={false}
          testID="MoviePoster"
        >
          <View
            style={
              [
                {
                  "height": 170,
                  "position": "relative",
                  "width": 110,
                },
                false,
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 100,
                  "opacity": 1,
                  "padding": 6,
                  "position": "absolute",
                  "right": 6,
                  "top": 6,
                  "zIndex": 10,
                }
              }
            />
            <Image
              resizeMode="cover"
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w440_and_h660_face/ngl2FKBlU4fhbdsrtdom9LVLBXw.jpg",
                }
              }
              style={
                [
                  {
                    "borderRadius": 8,
                    "height": 170,
                    "width": 110,
                  },
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "padding": 16,
              "paddingRight": 0,
            }
          }
          testID="MoviDetail"
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 14,
                "fontWeight": "400",
                "marginBottom": 8,
              }
            }
          >
            Ant-Man and the Wasp: Quantumania
          </Text>
          <Text
            style={
              {
                "color": "#757575",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          />
          <Text
            style={
              {
                "color": "#757575",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          >
            2023
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          [
            {
              "justifyContent": "flex-start",
              "marginBottom": 16,
            },
            {
              "flexDirection": "row",
            },
          ]
        }
        testID="MovieCard"
      >
        <View
          style={false}
          testID="MoviePoster"
        >
          <View
            style={
              [
                {
                  "height": 170,
                  "position": "relative",
                  "width": 110,
                },
                false,
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 100,
                  "opacity": 1,
                  "padding": 6,
                  "position": "absolute",
                  "right": 6,
                  "top": 6,
                  "zIndex": 10,
                }
              }
            />
            <Image
              resizeMode="cover"
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w440_and_h660_face/qNBAXBIQlnOThrVvA6mA2B5ggV6.jpg",
                }
              }
              style={
                [
                  {
                    "borderRadius": 8,
                    "height": 170,
                    "width": 110,
                  },
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "padding": 16,
              "paddingRight": 0,
            }
          }
          testID="MoviDetail"
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 14,
                "fontWeight": "400",
                "marginBottom": 8,
              }
            }
          >
            The Super Mario Bros. Movie
          </Text>
          <Text
            style={
              {
                "color": "#757575",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          />
          <Text
            style={
              {
                "color": "#757575",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          >
            2023
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          [
            {
              "justifyContent": "flex-start",
              "marginBottom": 16,
            },
            {
              "flexDirection": "row",
            },
          ]
        }
        testID="MovieCard"
      >
        <View
          style={false}
          testID="MoviePoster"
        >
          <View
            style={
              [
                {
                  "height": 170,
                  "position": "relative",
                  "width": 110,
                },
                false,
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 100,
                  "opacity": 1,
                  "padding": 6,
                  "position": "absolute",
                  "right": 6,
                  "top": 6,
                  "zIndex": 10,
                }
              }
            />
            <Image
              resizeMode="cover"
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w440_and_h660_face/2VK4d3mqqTc7LVZLnLPeRiPaJ71.jpg",
                }
              }
              style={
                [
                  {
                    "borderRadius": 8,
                    "height": 170,
                    "width": 110,
                  },
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "padding": 16,
              "paddingRight": 0,
            }
          }
          testID="MoviDetail"
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 14,
                "fontWeight": "400",
                "marginBottom": 8,
              }
            }
          >
            Shazam! Fury of the Gods
          </Text>
          <Text
            style={
              {
                "color": "#757575",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          />
          <Text
            style={
              {
                "color": "#757575",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          >
            2023
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          [
            {
              "justifyContent": "flex-start",
              "marginBottom": 16,
            },
            {
              "flexDirection": "row",
            },
          ]
        }
        testID="MovieCard"
      >
        <View
          style={false}
          testID="MoviePoster"
        >
          <View
            style={
              [
                {
                  "height": 170,
                  "position": "relative",
                  "width": 110,
                },
                false,
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#ffffff",
                  "borderRadius": 100,
                  "opacity": 1,
                  "padding": 6,
                  "position": "absolute",
                  "right": 6,
                  "top": 6,
                  "zIndex": 10,
                }
              }
            />
            <Image
              resizeMode="cover"
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w440_and_h660_face/t6HIqrRAclMCA60NsSmeqe9RmNV.jpg",
                }
              }
              style={
                [
                  {
                    "borderRadius": 8,
                    "height": 170,
                    "width": 110,
                  },
                  false,
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "padding": 16,
              "paddingRight": 0,
            }
          }
          testID="MoviDetail"
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontSize": 14,
                "fontWeight": "400",
                "marginBottom": 8,
              }
            }
          >
            Avatar: The Way of Water
          </Text>
          <Text
            style={
              {
                "color": "#757575",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          />
          <Text
            style={
              {
                "color": "#757575",
                "fontSize": 14,
                "marginBottom": 5,
              }
            }
          >
            2022
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
